### PR TITLE
CI using GitHub Actions

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

**Review by streeve — COMMENTED** [review]
We need to add gtest to the CMake, right? Can we point to the one in Cabana? Or do we need to pull down the tarball?

**Comment by junghans** [comment]
@streeve ready for your review, see my comment for discussion?

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
```
Comment (junghans) [inline .github/workflows/CI.yml:8]: As we are building on top of Cabana master, we might want to add a cron build here as well. Once a week?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+jobs:
+  CI:
+    strategy:
+      matrix:
+        backend: ["OPENMP"]
```
Comment (junghans) [inline .github/workflows/CI.yml:16]: Somehow `OPENMP` is hard-coded in some test and hence build just `SERIAL` fails. Not sure if worth fixing.

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Build kokkos
+        working-directory: kokkos
+        run: |
+          cmake -B build -DCMAKE_INSTALL_PREFIX=$HOME/kokkos -DKokkos_CXX_STANDARD=14 -DKokkos_ENABLE_${{ matrix.backend }}=ON
+          cmake --build build --parallel 2
+          cmake --install build
+      - name: Build Cabana
+        working-directory: Cabana
+        run: |
+          cmake -B build -DCMAKE_INSTALL_PREFIX=$HOME/Cabana -DCMAKE_PREFIX_PATH="$HOME/kokkos" -DCabana_REQUIRE_${{ matrix.backend }}=ON -DCabana_REQUIRE_MPI=ON -DCabana_ENABLE_CAJITA=OFF
```
Comment (junghans) [inline .github/workflows/CI.yml]: We can drop `-DCabana_REQUIRE_MPI=ON` once https://github.com/ECP-copa/Cabana/pull/272 is merged.

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        run: sudo apt-get install gfortran openmpi-bin libopenmpi-dev
+      - name: Checkout code
+        uses: actions/checkout@v2.2.0
+      - name: Checkout kokkos
+        uses: actions/checkout@v2.2.0
+        with:
+          repository: kokkos/kokkos
+          ref: 3.1.00
+          path: kokkos
+      - name: Checkout Cabana
```
Comment (junghans) [inline .github/workflows/CI.yml]: Maybe we should re-order the checkout and build blocks for better readability.

**Comment by junghans** [comment]
> We need to add gtest to the CMake, right? Can we point to the one in Cabana? Or do we need to pull down the tarball?

I seems `Cabana-fortran-examples` doesn't use gtest at all. Remember we aren't testing `Cabana` in this CI, but just `Cabana-fortran-examples`.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (streeve) [reply .github/workflows/CI.yml:8]: Sounds good to me

File: .github/workflows/CI.yml (hunk 2 above)
Comment (streeve) [reply .github/workflows/CI.yml:16]: I think adding `-DUSE_OMP=0` would fix `SERIAL`

File: .github/workflows/CI.yml (hunk 4 above)
Comment (streeve) [reply .github/workflows/CI.yml]: Agreed - Kokkos, then Cabana

File: .github/workflows/CI.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        working-directory: Cabana
+        run: |
+          cmake -B build -DCMAKE_INSTALL_PREFIX=$HOME/Cabana -DCMAKE_PREFIX_PATH="$HOME/kokkos" -DCabana_REQUIRE_${{ matrix.backend }}=ON
+          cmake --build build --parallel 2
+          cmake --install build
+      - name: Build
+        run: |
+          cmake -B build -DCMAKE_PREFIX_PATH="$HOME/kokkos;$HOME/Cabana"
+          cmake --build build --parallel 2
+          cd build && ctest --output-on-failure
```
Comment (streeve) [inline .github/workflows/CI.yml:51]: There are no unit tests, but the tutorial is set up to use gtest to make sure it least runs. Seems worth keeping

**Review by streeve — CHANGES_REQUESTED** [review]
Looks good, but I think it's easy enough to add serial and gtest

File: .github/workflows/CI.yml (hunk 5 above)
Comment (junghans) [reply .github/workflows/CI.yml:51]: Huh? `git grep -i gtest` returns nothing, where do we depend on gtest?

Comment (streeve) [reply .github/workflows/CI.yml:51]: You're right - it's missing in the top level CMakeLists, but all the tutorials already `add_test`. I should have included it in the previous PR

File: .github/workflows/CI.yml (hunk 1 above)
Comment (junghans) [reply .github/workflows/CI.yml:8]: Done.

File: .github/workflows/CI.yml (hunk 2 above)
Comment (junghans) [reply .github/workflows/CI.yml:16]: Maybe in a later PR.

File: .github/workflows/CI.yml (hunk 4 above)
Comment (junghans) [reply .github/workflows/CI.yml]: Done.

**Comment by junghans** [comment]
@streeve unless you want to fix the backend stuff this is ready to be merged.

**Review by streeve — APPROVED** [review]

**Pull request merged**